### ISSUE TITLE
Extend Update support for ElastiCache Global Replication Group

### ISSUE THREAD

**Issue by gdavison** [hashicorp/terraform-provider-aws#17696]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The PRs #15885 and #17615 add basic support for ElastiCache Global Replication Groups. However, only the `global_replication_group_description ` field can be updated. The ElastiCache API allows updating `AutomaticFailoverEnabled`, `CacheNodeType`, `CacheParameterGroupName`, and `EngineVersion` once a Global Replication Group has been created.

Add the ability to modify these parameters during an Update operation and call the Update operation as part of the Create operation.

- [x] Engine version #25077
- [x] Node type #27134
- [x] Automatic failover enabled #27134

### Affected Resource

* aws_elasticache_global_replication_group

### References

* #12708

**Comment by marinsalinas** [comment]
Hey @gdavison running into some issues when updating some of the parameters from `replication_group` that belongs to a `global_replication_group`, such as `replicas_per_node_group`. As far as I understand in the description, some of the attributes should be changed at `global_replication_group` level, but I think the replicas per node should be changed at the regional replication group level.


Here is the error we are getting:

```
Error: error updating ElastiCache Replication Group (dbtfmodules-test-redis): error requesting modification: InvalidParameterValue: Cluster [dbtfmodules-test-redis] is part of a global cluster [ldgnf-dbtfmodules-test-redis]. Request rejected.
	status code: 400, request id: 9e9912af-dab2-409d-96c9-7494bc55f074
```

Basically, we are using the same configuration from the exaple.
cc @bflad @maryelizbeth  should we create another ticket or we can track on this ticket since it is related?

**Comment by gdavison** [comment]
Hi @marinsalinas. Can you create a separate issue for that? It's a related issue, but will need to be solved separately. In your issue, can you also show which parameters are changing?

**Comment by marinsalinas** [comment]
Hello @gdavison, we are still having issues when trying to update a global replication group similarly we had on https://github.com/hashicorp/terraform-provider-aws/issues/19929

The parameter we wanted to update is the engine version.
